cpp/benchmarks/bench_simulation.cpp

```diff
@@ -0,0 +1,56 @@
+/* 
+* Copyright (C) 2020-2021 German Aerospace Center (DLR-SC)
+*
+* Authors: Rene Schmieding
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "bench_core.h"
+#include "bench_model.h"
+
+// google's benchmark templates require a function in the global scope, hence the following "using" statement
+using mio::benchmark::simulation;
+// set config path. the layout of the file depends on the initializer struct used to read it
+const char configpath[] = "benchmarks/simulation_benchmark.config";
+// specify a initilaizer which can read a config file for benchmark parameters.
+// the 'filepath' template cannot be a string literal.
+using init = mio::benchmark::simulation_file_init<configpath, mio::benchmark::model::SecirAgeresDampings>;
+
+// dummy runs to avoid large effects of cpu scaling on times of actual benchmarks
+BENCHMARK_TEMPLATE(simulation, mio::RKIntegratorCore, init)

```

Comment:
maybe using values and arguments (https://github.com/google/benchmark/blob/main/docs/user_guide.md#passing-arguments) would have been easier to understand for the parametrization? maybe not for the integrator, but surely for the model initialization. 

No need to redo it now, but in general, I think values are nicer than templates if possible. You wouldn't have to worry about string literals in templates for example...